"panic: assignment to entry in nil map" When assigning Policy to group

I can reproduce the issue posted on slack with the new Version RELEASE.2020-05-28T23-29-21Z which refuses to start when you have a policy attached to a group.

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/minio/minio/cmd.(*IAMObjectStore).loadAll(0xc0000adf40, 0x64c6d60, 0xc0002a6400, 0xc0004ac3c0, 0x0, 0x0)
	/Users/anton/go/src/github.com/minio/minio/cmd/iam-object-store.go:493 +0x1230
github.com/minio/minio/cmd.(*IAMSys).Init(0xc0004ac3c0, 0x64c6d60, 0xc0002a6400, 0x6505200, 0xc0006de700, 0x0, 0x0)
	/Users/anton/go/src/github.com/minio/minio/cmd/iam.go:427 +0x150
github.com/minio/minio/cmd.initAllSubsystems(0x6505200, 0xc0006de700, 0x0, 0x0)
	/Users/anton/go/src/github.com/minio/minio/cmd/server-main.go:326 +0x960
github.com/minio/minio/cmd.initSafeMode(0x0, 0x0)
	/Users/anton/go/src/github.com/minio/minio/cmd/server-main.go:238 +0x8f2
github.com/minio/minio/cmd.serverMain(0xc0000c8b60)
	/Users/anton/go/src/github.com/minio/minio/cmd/server-main.go:527 +0x1440
github.com/minio/cli.HandleAction(0x5c9be80, 0x609b278, 0xc0000c8b60, 0x0, 0x0)
	/Users/anton/go/pkg/mod/github.com/minio/cli@v1.22.0/app.go:500 +0x1b1
github.com/minio/cli.Command.Run(0x600a7ba, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x60324fd, 0x1b, 0x0, ...)
	/Users/anton/go/pkg/mod/github.com/minio/cli@v1.22.0/command.go:225 +0xdbe
github.com/minio/cli.(*App).Run(0xc0000c81a0, 0xc0000bc090, 0x3, 0x3, 0x0, 0x0)
	/Users/anton/go/pkg/mod/github.com/minio/cli@v1.22.0/app.go:261 +0xbf0
github.com/minio/minio/cmd.Main(0xc0000bc090, 0x3, 0x3)
	/Users/anton/go/src/github.com/minio/minio/cmd/main.go:169 +0xb0
main.main()
	/Users/anton/go/src/github.com/minio/minio/main.go:37 +0x45
```
## Expected Behavior
Do not panic
## Current Behavior
panic with stack trace above

## Possible Solution
Haven't researched yet

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->

1. mc admin user add localhost newuser userpass123
> Added user `newuser` successfully.
2. mc admin group add localhost newgroup newuser
> Added members {newuser} to group newgroup successfully.
3. mc admin policy set localhost readonly group=newgroup
> -- panic

## Context
This does not happen for Policies assigned to users 

`mc admin policy set localhost readonly user=newuser`

## Regression
Probably introduced in #9713 

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used (`minio version`): RELEASE.2020-05-28T23-29-21Z
* Environment name and version (e.g. nginx 1.9.1): - 
* Server type and version: -
* Operating System and version (`uname -a`): Darvin
* Link to your project:

--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -1798,6 +1798,7 @@ func NewIAMSys() *IAMSys {
 		iamUsersMap:             make(map[string]auth.Credentials),
 		iamPolicyDocsMap:        make(map[string]iampolicy.Policy),
 		iamUserPolicyMap:        make(map[string]MappedPolicy),
+		iamGroupPolicyMap:       make(map[string]MappedPolicy),
 		iamGroupsMap:            make(map[string]GroupInfo),
 		iamUserGroupMemberships: make(map[string]set.StringSet),
 	}

